data/json/items/gun/obsolete.json

```diff
@@ -256,5 +256,45 @@
     ],
     "modes": [ [ "DEFAULT", "semi-auto", 1 ], [ "AUTO", "auto", 4 ] ],
     "flags": [ "OBSOLETE", "NO_TURRET" ]
+  },
+  {
+    "id": "deagle_44",
```

Comment:
fyi you don't need an obsoletion if you migrate a weapon (you didn't)
```json
  {
    "id": "deagle_44",
    "type": "MIGRATION",
    "replace": "modular_deagle"
  },
```
Migration should be put somewhere in `data/json/obsoletion/migration_items.json`

Reply:
I wan under the impression migrations with gunmods didn't work, so I went with obsoletion like the original batch of modular guns did.  I'll change that up.

Reply:
it works, even transfer gunmods properly, but it will be affected with #70156, but it will be fixed sooner or later anyway